### PR TITLE
[Keymap] merge jdelkins userspace and associated keymaps

### PR DESCRIPTION
**Pull request by jdelkins**
## Description

Merge my userspace and keymaps. For a couple of keyboards, I have an alternate layout which I also want to use with my userspace, hence the users/jdelkins_ss -> users/jdelkins symlink (_ss is for split space). If there is a better way to handle these cases, I'd love to know.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include "jdelkins.h"
+#include "rgblight.h"
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:21]: This shouldn't be needed. 
```suggestion
```

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    }
+    ctl_state = 0;
+}
+
+void g_finished(qk_tap_dance_state_t *state, void *user_data) {
+    switch (cur_dance(state)) {
+        case SINGLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_END);
+            unregister_code(KC_LCTL);
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:61]:
```suggestion
            tap_code16(C(KC_END));
```

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_HOME);
+            unregister_code(KC_LCTL);
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:66]:
```suggestion
            tap_code16(C(KC_HOME));
```

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        SEQ_TWO_KEYS(KC_SCLN, KC_K) {
+            send_secret_string(4);
+        }
+        SEQ_TWO_KEYS(KC_SCLN, KC_L) {
+            send_secret_string(5);
+        }
+        SEQ_ONE_KEY(KC_C) {
+            register_code(KC_LCTL);
+            tap_code(KC_C);
+            unregister_code(KC_LCTL);
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:256]:
```suggestion
            tap_code16(C(KC_C));
```

File: keyboards/dz60/keymaps/jdelkins/rules.mk [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+LEADER_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+CONSOLE_ENABLE = no
+NKRO_ENABLE = no
+TAP_DANCE_ENABLE = yes
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins/rules.mk:21]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/dz60/keymaps/jdelkins_ss/keymap.c [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include "jdelkins.h"
+#include "rgblight.h"
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins_ss/keymap.c:21]:
```suggestion
```

File: keyboards/dz60/keymaps/jdelkins_ss/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+LEADER_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+CONSOLE_ENABLE = no
+NKRO_ENABLE = no
+TAP_DANCE_ENABLE = yes
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/dz60/keymaps/jdelkins_ss/rules.mk:21]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 8]
```diff
@@ -0,0 +1,514 @@
...
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include "jdelkins.h"
+#include "rgb_matrix.h"
```
Comment (drashna) [inline keyboards/massdrop/alt/keymaps/jdelkins/keymap.c:21]:
```suggestion
```

File: keyboards/massdrop/alt/keymaps/jdelkins_ss/keymap.c [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include "rgb_matrix.h"
```
Comment (drashna) [inline keyboards/massdrop/alt/keymaps/jdelkins_ss/keymap.c:20]:
```suggestion
```

File: keyboards/planck/keymaps/jdelkins/config.h [hunk 10]
```diff
@@ -0,0 +1,42 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/jdelkins/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/jdelkins/rules.mk [hunk 11]
```diff
@@ -0,0 +1,5 @@
+SRC += muse.c
+LEADER_ENABLE = yes
+TAP_DANCE_ENABLE = yes
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/planck/keymaps/jdelkins/rules.mk:4]:
```suggestion
LTO_ENABLE = yes
```

File: users/jdelkins_ss [hunk 12]
```diff
@@ -0,0 +1 @@
+jdelkins
```
Comment (drashna) [inline users/jdelkins_ss:1]: Is this file needed? 
If not, it should be removed.

**Review by drashna — CHANGES_REQUESTED** [review]

File: users/jdelkins_ss (hunk 12 above)
Comment (jdelkins) [reply users/jdelkins_ss:1]: This is a symlink to my userspace. It is needed unless there is another straightforward way to include my `jdelkins` userspace from my `jdelkins_ss` keymaps?

Comment (drashna) [reply users/jdelkins_ss:1]: Missed that in the comments.  And yes, there is, actually. 
https://docs.qmk.fm/#/feature_userspace?id=override-default-userspace

Specifically, if you add this to the rules.mk for the layouts:
```makefile
USER_NAME := jdelkins
```

This overrides the userspace that is used.

Comment (jdelkins) [reply users/jdelkins_ss]: That's fantastic. Thanks!

**Comment by jdelkins** [comment]
@drashna Thank you for your review. All suggested changes are committed.

File: users/jdelkins/jdelkins.c [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+            if (!record->event.pressed) {
+                clear_oneshot_layer_state(ONESHOT_OTHER_KEY_PRESSED);
+                send_secret_string(keycode - KC_SECRET_1);
+            }
+            return false;
+            break;
+#endif
+    }
+
+    return process_record_keymap(keycode, record);
```
Comment (drashna) [inline users/jdelkins/jdelkins.c:246]: This may be better at the begining of the function, as: 
```c
    if (!process_record_keymap(keycode, record)) { return false; }
```

This way, if the keymap function returns false, it stops the processing of further keycodes, which is more in line with the rest.

File: users/jdelkins/jdelkins.h [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+        uint32_t raw;
+        struct {
+                bool system_mac :1;
+        };
+} user_config_t;
+
+extern user_config_t user_config;
+
+static inline void do_mac_key(uint16_t norm_key, uint16_t mac_key, keyrecord_t *record) {
+    uint16_t key = user_config.system_mac ? mac_key : norm_key;
```
Comment (drashna) [inline users/jdelkins/jdelkins.h:62]: you could also use `keymap_config.lctrl_lgui_swap`, as this is the CG_SWAP stuff, which is very useful on mac. 

This is just a suggestion, though. I personally find it easier to use this, as it globally available.

File: users/jdelkins/jdelkins.h [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+#define MODS_SHIFT (get_mods() & MOD_MASK_SHIFT)
+#define MODS_CTRL  (get_mods() & MOD_MASK_CTRL)
+#define MODS_ALT   (get_mods() & MOD_MASK_ALT)
+#define MODS_GUI   (get_mods() & MOD_MASK_GUI)
+
+#define MY_CAPS LCTL_T(KC_CAPS)
+#define MY_SPC  LT(_FUNC, KC_SPC)
+
+#define NUMLOCK_ON (host_keyboard_leds() & (1<<USB_LED_NUM_LOCK))
+#define CAPSLOCK_ON (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK))
```
Comment (drashna) [inline users/jdelkins/jdelkins.h:124]:
```suggestion
#define NUMLOCK_ON host_keyboard_led_state().num_lock
#define CAPSLOCK_ON host_keyboard_led_state().caps_lock
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: users/jdelkins/jdelkins.h [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+// Secrets
+
+#ifndef NO_SECRETS
```
[the remark this answers is not in the document]
Comment (jdelkins) [reply users/jdelkins/jdelkins.h:24]: Not sure what you mean. Are you saying move the whole `#ifndef` block to jdelkins.c? I use `send_secret_string` in some keymaps and `SEC1`..`SEC6` are defined differently based on the `DO_SECRETS` macro. I could move the `#include "secrets.h"` part into the .c file I suppose, if that's what you mean.

File: users/jdelkins/jdelkins.h (hunk 14 above)
Comment (jdelkins) [reply users/jdelkins/jdelkins.h:62]: If it's the same to you, I think I'd prefer to keep it this way. I don't use bootmagic and using `keymap_config` instead of `user_config` without bootmagic doesn't get me anything.

File: users/jdelkins/jdelkins.h [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
```
[the remark this answers is not in the document]
Comment (jdelkins) [reply users/jdelkins/jdelkins.h]: Done, thanks

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 18]
```diff
@@ -0,0 +1,513 @@
...
+                if (!RGB_IS_NULL(*m))
+                    rgb_matrix_set_color(i, m->r, m->g, m->b);
+            }
+            break;
+    }
+}
+
+// Runs just one time when the keyboard initializes.
+void matrix_init_keymap(void) {
+    set_rgb_layer(_QWERTY);
```
[the remark this answers is not in the document]
Comment (jdelkins) [reply keyboards/massdrop/alt/keymaps/jdelkins/keymap.c]: Done, thanks

File: users/jdelkins/jdelkins.c (hunk 13 above)
Comment (jdelkins) [reply users/jdelkins/jdelkins.c]: Done, thanks

File: users/jdelkins/jdelkins.h (hunk 14 above)
Comment (drashna) [reply users/jdelkins/jdelkins.h:62]: That's absolutely fine!  This was much more of a suggestion, based on usage.

File: users/jdelkins/jdelkins.h (hunk 16 above)
[the remark this answers is not in the document]
Comment (drashna) [reply users/jdelkins/jdelkins.h:24]: The keycodes would need to remain, yeah, but the rest of the block could be moved, in theory.  But this is more of a nitpick/personal preference.

**Review by drashna — APPROVED** [review]

File: users/jdelkins/jdelkins.h (hunk 16 above)
[the remark this answers is not in the document]
Comment (jdelkins) [reply users/jdelkins/jdelkins.h]: Okay, done. I like it better this way also.

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 19]
```diff
@@ -0,0 +1,282 @@
...
+        // custom handle KC_GESC to emulate auto shift on it
+        case KC_GESC:
+            if (get_autoshift_state()) {
+                if (record->event.pressed)
+                    gesc_timer = timer_read();
+                else {
+                    if (timer_elapsed(gesc_timer) > AUTO_SHIFT_TIMEOUT || MODS_SHIFT) {
+                        register_code(KC_LSFT);
+                        tap_code(KC_GRV);
+                        unregister_code(KC_LSFT);
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:269]:
```suggestion
                        tap_code16(S(KC_GRV));
```

File: keyboards/dz60/keymaps/jdelkins/keymap.c [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+                    } else if (MODS_GUI)
+                        tap_code(KC_GRV);
+                    else
+                        tap_code(KC_ESC);
+                }
+                return false;
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/jdelkins/keymap.c:278]:
```suggestion
```

File: keyboards/dz60/keymaps/jdelkins/rules.mk [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/jdelkins/rules.mk:15]:
```suggestion
```

File: keyboards/dz60/keymaps/jdelkins_ss/keymap.c [hunk 22]
```diff
@@ -0,0 +1,374 @@
...
+    switch (cur_dance(state)) {
+        case SINGLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_END);
+            unregister_code(KC_LCTL);
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_HOME);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/jdelkins_ss/keymap.c:68]:
```suggestion
            tap_code16(C(KC_END));
            break;
        case DOUBLE_TAP:
            tap_code16(C(KC_HOME));
```

File: keyboards/dz60/keymaps/jdelkins_ss/rules.mk [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/jdelkins_ss/rules.mk:15]:
```suggestion
```

File: keyboards/massdrop/alt/keymaps/jdelkins/rules.mk [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins/rules.mk:15]:
```suggestion
```

File: keyboards/massdrop/alt/keymaps/jdelkins_ss/config.h [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define RGB_MATRIX_STARTUP_SPD 20
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins_ss/config.h:18]:
```suggestion
#pragma once

#define RGB_MATRIX_STARTUP_SPD 20
```

File: keyboards/massdrop/alt/keymaps/jdelkins_ss/keymap.c [hunk 26]
```diff
@@ -0,0 +1,280 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+#include <print.h>
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins_ss/keymap.c:20]:
```suggestion
#include "print.h"
```

File: keyboards/massdrop/alt/keymaps/jdelkins_ss/rules.mk [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins_ss/rules.mk:15]:
```suggestion
```

File: keyboards/planck/keymaps/jdelkins/rules.mk [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/jdelkins/rules.mk:15]:
```suggestion
```

File: users/jdelkins/jdelkins.c [hunk 29]
```diff
@@ -0,0 +1,286 @@
...
+#elif defined(BOOTLOADER_DFU) || defined(BOOTLOADER_LUFA_DFU) || defined(BOOTLOADER_QMK_DFU)
+                            ":dfu "
+#elif defined(BOOTLOADER_HALFKAY)
+                            ":teensy "
+#elif defined(BOOTLOADER_CATERINA)
+                            ":avrdude "
+#else
+                            ":doflash "
+#endif
+                            SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline users/jdelkins/jdelkins.c:242]:
```suggestion
                    SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP ":flash\n");
```

File: users/jdelkins/rules.mk [hunk 30]
```diff
@@ -0,0 +1,25 @@
...
+ # (at your option) any later version.
+
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
```
Comment (fauxpark) [inline users/jdelkins/rules.mk:15]:
```suggestion
```

File: keyboards/planck/keymaps/jdelkins/keymap.c [hunk 31]
```diff
@@ -0,0 +1,429 @@
...
+    }
+    ctl_state = 0;
+}
+
+void g_finished(qk_tap_dance_state_t *state, void *user_data) {
+    switch (cur_dance(state)) {
+        case SINGLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_END);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/jdelkins/keymap.c:69]:
```suggestion
            tap_code16(C(KC_END));
```

File: keyboards/planck/keymaps/jdelkins/keymap.c [hunk 32]
```diff
@@ -0,0 +1,429 @@
...
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_HOME);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/jdelkins/keymap.c:74]:
```suggestion
            tap_code16(C(KC_HOME));
```

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 33]
```diff
@@ -0,0 +1,510 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "jdelkins.h"
+#include <print.h>
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins/keymap.c:19]:
```suggestion
#include "print.h"
```

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 34]
```diff
@@ -0,0 +1,510 @@
...
+    }
+    ctl_state = 0;
+}
+
+void g_finished(qk_tap_dance_state_t *state, void *user_data) {
+    switch (cur_dance(state)) {
+        case SINGLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_END);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins/keymap.c:71]:
```suggestion
            tap_code16(C(KC_END));
```

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 35]
```diff
@@ -0,0 +1,510 @@
...
+            break;
+        case DOUBLE_TAP:
+            register_code(KC_LCTL);
+            tap_code(KC_HOME);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins/keymap.c:76]:
```suggestion
            tap_code16(C(KC_HOME));
```

File: keyboards/massdrop/alt/keymaps/jdelkins/keymap.c [hunk 36]
```diff
@@ -0,0 +1,510 @@
...
+#ifdef AUTO_SHIFT_ENABLE
+            if (get_autoshift_state()) {
+                static uint16_t gesc_timer;
+                if (record->event.pressed)
+                    gesc_timer = timer_read();
+                else {
+                    if (timer_elapsed(gesc_timer) > AUTO_SHIFT_TIMEOUT || MODS_SHIFT) {
+                        register_code(KC_LSFT);
+                        tap_code(KC_GRV);
+                        unregister_code(KC_LSFT);
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/jdelkins/keymap.c:376]:
```suggestion
                        tap_code16(S(KC_GRV));
```

File: keyboards/planck/keymaps/jdelkins/keymap.c [hunk 37]
```diff
@@ -0,0 +1,429 @@
...
+        SEQ_TWO_KEYS(KC_SCLN, KC_K) {
+            send_secret_string(4);
+        }
+        SEQ_TWO_KEYS(KC_SCLN, KC_L) {
+            send_secret_string(5);
+        }
+        SEQ_ONE_KEY(KC_C) {
+            register_code(KC_LCTL);
+            tap_code(KC_C);
+            unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/planck/keymaps/jdelkins/keymap.c:416]:
```suggestion
            tap_code16(C(KC_C));
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: users/jdelkins/jdelkins.c (hunk 29 above)
Comment (jdelkins) [reply users/jdelkins/jdelkins.c]: Great suggestion. I have one keyboard (massdrop/alt) that doesn't support :flash so I still need to special-case that one.

Comment (fauxpark) [reply users/jdelkins/jdelkins.c]: I think you should be able to set `PROGRAM_CMD` in rules.mk to allow `:flash` to run that if no other bootloader is specified, rather than defining your own flash target.

Comment (jdelkins) [reply users/jdelkins/jdelkins.c]: Okay, that's a good idea. However, this requires a small change to `tmk_core/arm_atsam.mk` to make use of `PROGRAM_CMD` which I'll submit as a separate PR since it's outside of the keyboard tree

**Comment by jdelkins** [comment]
@fauxpark I think I included all of your requested changes. LMK if you request anything else. Thanks :)

**Comment by jdelkins** [comment]
Should I close this and re-open a squashed PR

**Comment by jdelkins** [comment]
Hi, this has been hanging around for a while. I think I have made all of the requested changes. It is a lot of commits, however. Should I close this and re-open a squashed PR to clean it up?

**Comment by drashna** [comment]
> Hi, this has been hanging around for a while. I think I have made all of the requested changes. It is a lot of commits, however. Should I close this and re-open a squashed PR to clean it up?

No need.  We usually squash when merging, Especially for PRs like this.

File: users/jdelkins/jdelkins.c [hunk 38]
```diff
@@ -0,0 +1,274 @@
...
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (mac mode)");
+                    } else {
+                        SEND_STRING(QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION " (non-mac mode)");
+                    }
+                }
+                return false;
+            }
+            break;
+
+        case KB_BOOT:
```
Comment (drashna) [inline users/jdelkins/jdelkins.c:214]: You can use `RESET` here instead and replace the normal functionality of the keycode.

**Review by drashna — APPROVED** [review]

**Review by jdelkins — COMMENTED** [review]
all changes incorporated

**Review by tzarc — APPROVED** [review]
